service/http.directory/src/main/java/org/kaazing/gateway/service/http/directory/HttpDirectoryServiceHandler.java

```diff
@@ -104,6 +104,10 @@ void setPatterns(List<PatternCacheControl> patterns) {
         this.patterns = patterns;
     }
 
+    void emptyUrlCacheControlMap() {
+        urlCacheControlMap.clear();
+    }
```

Comment:
Cleaning up the resources is good, but I wonder why this is affecting the tests. Then when I see that urlCacheControlMap is static. So two questsions:
* I wonder why we need to keep it as static.
* urlCacheControlMap is accessed from multiple threads(say two clients are acessing the same directory service) , but currently it uses HashMap and is not thread-safe.